### PR TITLE
Query appointment by doctor

### PR DESCRIPTION
**Pull request by Kappaccinoh**
queries appointments by doctors NRIC - needs to match exactly

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/QueryDoctorAppointmentCommand.java
Comment (Alteqa) [inline src/main/java/seedu/address/logic/commands/QueryDoctorAppointmentCommand.java:1]: Very clean code and aligns with the rest of the query commands.

File: src/main/java/seedu/address/logic/parser/QueryDoctorAppointmentCommandParser.java
Comment (Alteqa) [inline src/main/java/seedu/address/logic/parser/QueryDoctorAppointmentCommandParser.java:1]: Clean and straightforward.

**Review by Alteqa — APPROVED** [review]
All looks good, nothing major to change.

File: src/main/java/seedu/address/logic/commands/QueryDoctorAppointmentCommand.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        return predicate.equals(otherQueryDoctorAppointmentCommand.predicate);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("predicate", predicate)
+                .toString();
+    }
+}
```
Comment (alfaloo) [inline src/main/java/seedu/address/logic/commands/QueryDoctorAppointmentCommand.java:66]: looks golden to me

File: src/main/java/seedu/address/logic/parser/QueryDoctorAppointmentCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, QueryDoctorAppointmentCommand.MESSAGE_USAGE));
+        }
+
+        String[] nameKeywords = trimmedArgs.split("\\s+");
+        logger.log(Level.INFO, "Name keywords: " + Arrays.toString(nameKeywords));
+
+        return new QueryDoctorAppointmentCommand(new AppointmentContainsDoctorPredicate(Arrays.asList(nameKeywords)));
+    }
+}
```
Comment (alfaloo) [inline src/main/java/seedu/address/logic/parser/QueryDoctorAppointmentCommandParser.java:37]: this looks good to me

File: src/main/java/seedu/address/model/appointment/AppointmentContainsDoctorPredicate.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        AppointmentContainsDoctorPredicate otherPredicate = (AppointmentContainsDoctorPredicate) other;
+        return keywords.equals(otherPredicate.keywords);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this).add("keywords", keywords).toString();
+    }
+}
```
Comment (alfaloo) [inline src/main/java/seedu/address/model/appointment/AppointmentContainsDoctorPredicate.java:50]: looks good for me

File: src/test/java/seedu/address/logic/commands/QueryDoctorAppointmentCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        assertEquals(expected, findCommand.toString());
+    }
+
+    /**
+     * Parses {@code userInput} into a {@code AppointmentContainsDoctorPredicate}.
+     */
+    private AppointmentContainsDoctorPredicate preparePredicate(String userInput) {
+        return new AppointmentContainsDoctorPredicate(Arrays.asList(userInput.split("\\s+")));
+    }
+}
```
Comment (alfaloo) [inline src/test/java/seedu/address/logic/commands/QueryDoctorAppointmentCommandTest.java:99]: nice and detailed tests

File: src/test/java/seedu/address/model/appointment/AppointmentContainsDoctorPredicateTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    @Test
+    public void toStringMethod() {
+        List<String> keywords = List.of("keyword1", "keyword2");
+        AppointmentContainsDoctorPredicate predicate = new AppointmentContainsDoctorPredicate(keywords);
+
+        String expected = AppointmentContainsDoctorPredicate.class.getCanonicalName()
+                + "{keywords=" + keywords + "}";
+        assertEquals(expected, predicate.toString());
+    }
+}
```
Comment (alfaloo) [inline src/test/java/seedu/address/model/appointment/AppointmentContainsDoctorPredicateTest.java:91]: I love these tests

**Pull request merged**